### PR TITLE
Add Sankey chart Plotly Adapter

### PR DESCRIPTION
**Pull request by Anush2303**
image

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: packages/charts/react-charting/package.json [hunk 1]
```diff
@@ -15,7 +15,6 @@
   "license": "MIT",
   "scripts": {
     "build": "just-scripts build",
-    "lint": "just-scripts lint",
```
Comment (AtishayMsft) [inline packages/charts/react-charting/package.json:18]: revert this change

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts [hunk 2]
```diff
@@ -88,6 +89,49 @@ export const transformPlotlyJsonToColumnProps = (jsonObj: any): IVerticalStacked
...
+      target: link.target[index],
+    }))
+    // eslint-disable-next-line @typescript-eslint/no-shadow
+    .filter(link => link.source !== link.target); // Filter out self-references(circular links)
+
+  const sankeyChartData = {
+    nodes: node.label.map((label: string, index: number) => ({
+      nodeId: index,
+      name: label,
+      color: node.color[index] || '#D3D3D3',
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:108]: use data viz palette

File: packages/charts/react-charting/package.json (hunk 1 above)
Comment (Anush2303) [reply packages/charts/react-charting/package.json]: Yes, was doing that.

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts (hunk 2 above)
Comment (Anush2303) [reply packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:108]: Okay

File: packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts [hunk 3]
```diff
@@ -88,6 +89,49 @@ export const transformPlotlyJsonToColumnProps = (jsonObj: any): IVerticalStacked
...
+  };
+
+  const width: number = layout?.width || 440;
+  const height: number = layout?.height || 220;
+  const styles: ISankeyChartProps['styles'] = {
+    root: {
+      fontSize: layout.font?.size,
+    },
+  };
+  const shouldResize: number = width + height;
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:121]: what does this do

Comment (Anush2303) [reply packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:121]: Property defined in ISankeyChartProps
/**
   * should chart resize when parent resize.
   */
  shouldResize?: number;

Comment (AtishayMsft) [reply packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:121]: but why you are adding width and height to get it.

Comment (AtishayMsft) [reply packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts:121]: ok. Kshitij would know then.

**Review by AtishayMsft — APPROVED** [review]
